lib/vehicle_factory.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+
+class VehicleFactory
+  attr_reader :vehicles
+
+  def initialize
+    @vehicles = []
+  end
+
+  def create_vehicles(database)
+    if database.first.keys.include?(:scofflaw_indicator)
```

Comment:
I like that you're letting the data work for you in determining which state/set of data you're working with.